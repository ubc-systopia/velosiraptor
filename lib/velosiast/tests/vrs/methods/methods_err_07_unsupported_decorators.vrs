
#[foo]
abstract fn foo() -> bool
    ensures true;

#[bar]
fn foo() -> bool
{
    true
}

#[foo, bar(baz)]
fn foo() -> bool
{
    true
}

#[remap(foo)]
fn foo() -> bool
{
    true
}