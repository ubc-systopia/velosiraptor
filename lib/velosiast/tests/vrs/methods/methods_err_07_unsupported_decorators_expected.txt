error: unsupported method property
     --> tests/vrs/methods/methods_err_07_unsupported_decorators.vrs:2:3
      |
    2 |  #[foo]
      |    ^^^ supported method properties are `invariant` and `remap`
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: unsupported method property
     --> tests/vrs/methods/methods_err_07_unsupported_decorators.vrs:6:3
      |
    6 |  #[bar]
      |    ^^^ supported method properties are `invariant` and `remap`
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: unsupported method property
     --> tests/vrs/methods/methods_err_07_unsupported_decorators.vrs:12:3
      |
   12 |  #[foo, bar(baz)]
      |    ^^^ supported method properties are `invariant` and `remap`
error: unsupported method property
     --> tests/vrs/methods/methods_err_07_unsupported_decorators.vrs:12:8
      |
   12 |  #[foo, bar(baz)]
      |         ^^^ supported method properties are `invariant` and `remap`
error: could not compile due to 2 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: method property `remap` doesn't support arguments
     --> tests/vrs/methods/methods_err_07_unsupported_decorators.vrs:18:9
      |
   18 |  #[remap(foo)]
      |          ^^^ remove these arguments
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
